pkg/apis/eventing/v1alpha1/channel_validation.go

```diff
@@ -34,9 +34,13 @@ func (cs *ChannelSpec) Validate() *apis.FieldError {
 		errs = errs.Also(apis.ErrMissingField("provisioner"))
 	}
 
-	for i, subscriber := range cs.Subscribers {
-		if subscriber.Call == nil && subscriber.Result == nil {
-			errs = errs.Also(apis.ErrMissingField("call", "result").ViaField(fmt.Sprintf("subscriber[%d]", i)))
+	if cs.Channelable != nil {
+		for i, subscriber := range cs.Channelable.Subscribers {
+			if subscriber.SinkableDomain == "" && subscriber.CallableDomain == "" {
+				fe := apis.ErrMissingField("sinkableDomain", "callableDomain")
+				fe.Details = "expected at least one of, got none"
```

Comment:
cc @n3wscott I think there's a helper for this?

Reply:
There is no helper for adding details to a Err* type method, there is a helper for setting the index.

it should be:

`errs = errs.Also(apis.ErrMissingField("sinkableDomain", "callableDomain").ViaFieldIndex("subscribers", i))`